trie/sync.go

```diff
@@ -85,30 +72,55 @@ func NewSyncPath(path []byte) SyncPath {
 	return SyncPath{hexToKeybytes(path[:64]), hexToCompact(path[64:])}
 }
 
-// SyncResult is a response with requested data along with it's hash.
-type SyncResult struct {
-	Hash common.Hash // Hash of the originally unknown trie node
-	Data []byte      // Data content of the retrieved node
+// nodeRequest represents a scheduled or already in-flight trie node retrieval request.
+type nodeRequest struct {
+	key  string // Key of the node data content to retrieve
+	path []byte // Merkle path leading to this node for prioritization
+	data []byte // Data content of the node, cached until all subtrees complete
+
+	parents  []*nodeRequest // Parent state nodes referencing this entry (notify all upon completion)
```

Comment:
Wondering if it can ever happen that we have multiple parents?

In a fast-sync hash-base-trie world, it can I guess happen that two subtries hash to the same root. But in a path based world, every node has one and only one parent, so it should not ever happen that we have >1 parent.

Reply:
fixed